### PR TITLE
CRI-O: add e2e branch tests

### PR DESCRIPTION
**Pull request by runcom**
Add branch tests to be used in postsubmits over at openshift/release https://github.com/openshift/release/pull/498

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
@mrunalp @stevekuznetsov PTAL

File: sjb/config/test_cases/test_branch_crio_e2e_rhel.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - PULL_BASE_SHA
+      - PULL_PULL_SHA
+  - type: "script"
+    title: "clone cri-o and check out the correct refs"
+    script: |-
+      cd /go/src/github.com/kubernetes-incubator/cri-o
+      git fetch origin
+      git checkout master
+      git branch -D target || true
+      git branch -D pr || true
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_branch_crio_e2e_rhel.yml]: can remove this

File: sjb/config/test_cases/test_branch_crio_e2e_rhel.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      git branch target "${PULL_BASE_SHA}"
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_branch_crio_e2e_rhel.yml]: if we can be in a detached HEAD we can also remove this

Comment (stevekuznetsov) [reply sjb/config/test_cases/test_branch_crio_e2e_rhel.yml]: and just checkout the ref

**Comment by runcom** [comment]
@stevekuznetsov took care of your comments :)

**Comment by stevekuznetsov** [comment]
@runcom do the XML generation in a new PR :)

**Comment by runcom** [comment]
> @runcom do the XML generation in a new PR :)

yup, I'm doing it in https://github.com/openshift/aos-cd-jobs/pull/945

**Pull request merged**